python/ray/tune/ray_trial_executor.py

```diff
@@ -101,15 +108,20 @@ def add(self, trial, actor):
         Args:
             trial (Trial): The trial corresponding to the future.
             actor (ActorHandle): Handle to the trainable to be stopped.
+            placement_group (PlacementGroup): Placement group to stop.
         """
         future = actor.stop.remote()
-        actor.__ray_terminate__.remote()
+
+        if placement_group:
+            remove_placement_group(placement_group)
+        else:
+            actor.__ray_terminate__.remote()
```

Comment:
If I do both I almost always run into a `SystemExit` error. However, even with this it sometimes shows up. Can we just get rid of the log message somehow? Or what's the best way here to terminate an ray actor and subsequently terminate the placement group?
cc @ericl 
```
(pid=50844) libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
(pid=50904) 2021-01-14 16:36:24,948	ERROR worker.py:390 -- SystemExit was raised from the worker
(pid=50904) Traceback (most recent call last):
(pid=50904)   File "python/ray/_raylet.pyx", line 570, in ray._raylet.task_execution_handler
(pid=50904)   File "python/ray/_raylet.pyx", line 434, in ray._raylet.execute_task
(pid=50904)   File "python/ray/_raylet.pyx", line 473, in ray._raylet.execute_task
(pid=50904)   File "python/ray/_raylet.pyx", line 476, in ray._raylet.execute_task
(pid=50904)   File "python/ray/_raylet.pyx", line 480, in ray._raylet.execute_task
(pid=50904)   File "python/ray/_raylet.pyx", line 432, in ray._raylet.execute_task.function_executor
(pid=50904)   File "/Users/kai/coding/ray/python/ray/function_manager.py", line 556, in actor_method_executor
(pid=50904)     return method(__ray_actor, *args, **kwargs)
(pid=50904)   File "/Users/kai/coding/ray/python/ray/tune/trainable.py", line 555, in stop
(pid=50904)     self.cleanup()
(pid=50904)   File "/Users/kai/coding/ray/python/ray/tune/function_runner.py", line 466, in cleanup
(pid=50904)     self._continue_semaphore.release()
(pid=50904)   File "/Users/kai/.pyenv/versions/3.7.7/lib/python3.7/threading.py", line 444, in release
(pid=50904)     self._cond.notify()
(pid=50904)   File "/Users/kai/.pyenv/versions/3.7.7/lib/python3.7/threading.py", line 352, in notify
(pid=50904)     waiter.release()
(pid=50904)   File "/Users/kai/coding/ray/python/ray/worker.py", line 387, in sigterm_handler
(pid=50904)     sys.exit(1)
(pid=50904) SystemExit: 1
```

Reply:
Maybe the alternative would be a `PlacementGroupCleanup` object that periodically checks if actors are offline and then removes the placement group. Thoughts?

Reply:
cc @rkooo567

Reply:
Are you saying you are seeing this log if you remove placement group and exit the actor right away?

Reply:
I'm seeing this log sometimes at the end of a tuning run. Sometimes I don't see it at all. If I call `__ray_terminate__` and then remove the placement group I see it every time (i.e. during a run once a placement group is removed)

Reply:
Hmm interesting. Can you create an issue with a simple reproduction? I will take a look at it soon. Is it a hard blocker?

Reply:
Not a hard blocker as placement groups won't be used by default yet. I'll try to get a simple repro script running tomorrow.

Reply:
Got it. I will mark p1 and work on it next sprint

Reply:
Thanks again for the repro :)!

Reply:
https://github.com/ray-project/ray/issues/13487

Thanks for looking into this!